mount.go

```diff
@@ -271,13 +271,23 @@ func virtioFSStorageHandler(storage pb.Storage, s *sandbox) (string, error) {
 
 // virtioBlkStorageHandler handles the storage for blk driver.
 func virtioBlkStorageHandler(storage pb.Storage, s *sandbox) (string, error) {
-	// Get the device node path based on the PCI address provided
-	// in Storage Source
-	devPath, err := getPCIDeviceName(s, storage.Source)
-	if err != nil {
-		return "", err
+
+	// If hot-plugged, get the device node path based on the PCI address else
+	// use the virt path provided in Storage Source
+	if strings.Contains(storage.Source, "/dev") {
```

Comment:
Please can you add some unit tests for this code? This is a very weak match (not anchored) so would match things like `/home/developer/foo.txt`. Can you add anchoring reliably to guarantee you only match on actual devices?

Reply:
Using HasPrefix() now instead of Contains. Added a few negative test cases to check the flow.